test/controllers/search_controller_test.rb

```diff
@@ -455,4 +455,391 @@ def source_filter_count(controller)
+        assert_select 'input#geobox-minLatitude', value: 60.0
+        assert_select 'input#geobox-maxLongitude', value: 78.2
+        assert_select 'input#geobox-maxLatitude', value: 80.0
+        assert_select 'input#geodistance-latitude', value: 36.1
+        assert_select 'input#geodistance-longitude', value: 62.6
+        assert_select 'input#geodistance-distance', value: '50mi'
+      end
+    end
+
+    test 'geospatial search can combine with basic and advanced inputs' do
+      VCR.use_cassette('geo all',
+                        allow_playback_repeats: true,
+                        match_requests_on: %i[method uri body]) do
+        query = {
+          q: 'hi',
+          title: 'hey',
+          citation: 'hello',
+          geobox: 'true',
+          geodistance: 'true',
+          geoboxMinLongitude: 40.5,
+          geoboxMinLatitude: 60.0,
+          geoboxMaxLongitude: 78.2,
+          geoboxMaxLatitude: 80.0,
+          geodistanceLatitude: 36.1,
+          geodistanceLongitude: 62.6,
+          geodistanceDistance: '50mi' 
+        }.to_query
+        get "/results?#{query}"
+        assert_response :success
+        assert_nil flash[:error]
+      end
+    end
+
+    test 'geobox arguments are non-nullable' do
+      query = {
+        geobox: 'true'
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "All bounding box fields are required."
+    end
+
+    test 'geodistance arguments are non-nullable' do
+      query = {
+        geodistance: 'true'
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "All geospatial distance fields are required."
+    end
+
+    test 'geobox param is required for geobox search' do
+      query = {
+        geoboxMinLongitude: 40.5,
+        geoboxMinLatitude: 60.0,
+        geoboxMaxLongitude: 78.2,
+        geoboxMaxLatitude: 80.0
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "A search term is required."
+    end
+
+    test 'geodistance param is required for geobox search' do
+      query = {
+        geodistanceLatitude: 36.1,
+        geodistanceLongitude: 62.6,
+        geodistanceDistance: '50mi' 
+      }.to_query
+      get "/results?#{query}"
+      assert_response :redirect
+      assert_equal flash[:error], "A search term is required."
+    end
+
+    test 'geobox lat/long cannot fall below min limit' do
+      low_query = {
+        geobox: 'true',
+        geoboxMinLongitude: -180.2,
+        geoboxMinLatitude: -90.2,
+        geoboxMaxLongitude: -180.1,
+        geoboxMaxLatitude: -90.1
+      }.to_query
+      get "/results?#{low_query}"
+      assert_response :redirect
+      assert_equal flash[:error], "Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0."
+    end
+
+    test 'geobox lat/long cannot exceed max limit' do
```

Comment:
I think the same issues with the minimum-value test are afflicting this test as well.

Valid, no errors:
https://timdex-ui-pi-gdt-136-lo-2ovnii.herokuapp.com/results?q=&geobox=true&geoboxMinLongitude=179&geoboxMinLatitude=89&geoboxMaxLongitude=180&geoboxMaxLatitude=90

Bumping either of the maximum values beyond the limit results in a 500 error, not a validation error.

EDIT - I think I found the cause - the validation clause being tested here requires both pairs of latitudes and longitudes to include an invalid value in order to return a failure.

Reply:
Yeah, I found the same issue when you mentioned the 500 errors in Slack. The current range validation logic can prevent it from erroring if it returns before it finds an invalid value.